core/src/main/java/org/apache/iceberg/data/avro/DataReader.java

```diff
@@ -36,6 +36,10 @@
 import org.apache.iceberg.types.Type;
 import org.apache.iceberg.types.Types;
 
+/**
+ * @deprecated will be removed in 2.0.0; use {@link PlannedDataReader} instead.
+ */
+@Deprecated
```

Comment:
Most of the changed files in this PR are to avoid using this class now that it is deprecated.